sr_hand_health_report/src/sr_hand_health_report/tactile_check.py

```diff
@@ -27,21 +27,43 @@ class TactileCheck(SrHealthReportCheck):
 
     _REASONABLE_RANGE = {"pst": [200, 1200], "bt_sp": [1800, 2400], "bt_2sp": [1800, 2400]}
     _FINGERS = ("FF", "MF", "RF", "LF", "TH")
+    TIMEOUT = 2
     PASSED_THRESHOLDS = {'connected': True, 'reasonable': True}
 
     def __init__(self, hand_side):
+        """
+            Initialize the TactileCheck object
+            @param hand_side: String indicating the side
+        """
         super().__init__(hand_side, self._FINGERS)
+        self._name = "Tactile"
         self._topic_name = f"/{self._hand_prefix}/tactile"
-        self._pass_conditions = {'connected': True, 'reasonable': True}
+        self._result = {'tactile': {}}
+
         try:
-            self._serial = rospy.get_param(f"/sr_hand_robot/{self._hand_prefix}/hand_serial")
+            serials = rospy.get_param("/diagnostic_aggregator/load_diagnostic_analyzer/hand_serials_list")
+            if len(serials) == 2:
+                self._serial = serials[0] if self._hand_prefix == 'rh' else serials[1]
+            else:
+                self._serial = serials[0]
         except KeyError:
+            pass
+
+        try:
+            self._serial
+        except Exception:
```

Comment:
Although that works, it is not really elegant and might catch errors you might not want in the future. Especially that by your condition above sel._serial will always exist.
I would advise you to initialise self._serial to None after self._result and then:
1) Try to get the param. If you can't then raise a specific exception
2) If you get the param, do set the value of self._serial
3) Potentially add any test you want before throwing another exception saying that no hand has been found (e.g. `if self._serial is None`)

Reply:
I wrapped it into:

```suggestion
        try:
            serials = rospy.get_param("/diagnostic_aggregator/load_diagnostic_analyzer/hand_serials_list")
            if len(serials) == 2:
                self._serial = serials[0] if self._hand_prefix == 'rh' else serials[1]
            else:
                self._serial = serials[0]
        except KeyError:
            rospy.logerr("No hand detected!")
            sys.exit(1)
```
If the param does not exist, then there will be the default "None" serial which is anyways 'redundant' as the check needs to fail/exit.

Reply:
Would this be fine with you?

Reply:
:+1: